pkg/generate/app/cmd/newapp.go

```diff
@@ -1109,11 +1109,32 @@ func optionallyValidateExposedPorts(config *AppConfig, repositories app.SourceRe
 	return nil
 }
 
-func exposedPortsAreNumeric(node *dockerfileparser.Node) error {
+func exposedPortsAreValid(node *dockerfileparser.Node) error {
+	allErrs := make([]error, 0)
+
 	for _, port := range dockerfileutil.LastExposedPorts(node) {
-		if _, err := strconv.ParseInt(port, 10, 32); err != nil {
-			return fmt.Errorf("could not parse %q: must be numeric", port)
+		errs := make([]string, 0)
```

Comment:
Don't we already have parse function for PORT/PROTOCOL somewhere?

Reply:
Updated